src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -617,6 +618,58 @@ public void InvalidMaxCPUCountSwitch4()
             });
         }
 
+        [Theory]
+        [InlineData("-getProperty:Foo;Bar", true, "EvalValue", false, false)]
+        [InlineData("-getProperty:Foo;Bar -t:Build", true, "TargetValue", false, false)]
+        [InlineData("-getItem:MyItem", false, "", true, false)]
+        [InlineData("-getItem:MyItem -t:Build", false, "", true, true)]
+        [InlineData("-getItem:WrongItem -t:Build", false, "", false, false)]
+        [InlineData("-getProperty:Foo;Bar -getItem:MyItem -t:Build", true, "TargetValue", true, true)]
+        [InlineData("-getProperty:Foo;Bar -getItem:MyItem", true, "EvalValue", true, false)]
+        public void ExecuteAppWithGetPropertyAndItem(string extraSwitch, bool fooPresent, string fooResult, bool itemIncludesAlwaysThere, bool itemIncludesTargetItem)
```

Comment:
Consider adding new tests with Verify, preserving the full JSON output.

Reply:
Ping on this.

Reply:
I don't understand how this adds value over what I currently have. I'm already verifying that the output includes the parts I want it to include and doesn't include a value I don't want it to include. If you go down deep enough, some of the metadata are ultimately based on hashing (a dictionary), which means that we can't guarantee that the output will look exactly as it currently does over time. Though I'd be surprised if it changes, I'd still prefer to avoid that risk.

Reply:
Since this is console UX, I think consistency and predictability is important. Order is a good consideration--I'd probably say to sort by key.

But if you feel very strongly about this, I think the current state meets the minimum bar.

Reply:
> I don't understand how this adds value over what I currently have.

This still applies for the test change.

As far as having a consistent/predictable order, the primary use case from my perspective is via automation, which doesn't care about the order.

Another ordering consideration is built-in metadata versus custom metadata. As-is, one of the three implementations explicitly separates custom metadata and built-in metadata and adds them separately. The other two instances do not guarantee that that will be the case, but many implementations do so implicitly, albeit not necessarily the same way. I think having custom metadata and built-in metadata in their own chunks is a valuable thing to include.